src/compiler/preprocessor.rs

```diff
@@ -140,3 +174,29 @@ pub fn preprocess(
 
     preprocess_(opts, includes, tocompile)
 }
+
+// Visit all files used during compilation.
+pub fn gather_dependencies(
+    opts: Rc<dyn CompilerOpts>,
+    real_input_path: &str,
+    file_content: &str,
+) -> Result<Vec<IncludeDesc>, CompileErr> {
+    let mut includes = Vec::new();
+    let loc = Srcloc::start(real_input_path);
+
+    let parsed = parse_sexp(loc.clone(), file_content.bytes()).map_err(|e| CompileErr(e.0, e.1))?;
```

Comment:
I suspect that if `CompileErr` would implement the `From` trait for the error type `parse_sexp()` returns, you would not need the `map_err()`

Reply:
Not a bad idea.  I'll do that.